core/src/main/java/com/linecorp/armeria/internal/server/annotation/AnnotatedServiceTypeUtil.java

```diff
@@ -82,6 +95,69 @@
                     .put("0", false)
                     .build();
 
+    /**
+     * Try to get a public static method with a single {@link String} argument in clazz.
+     * @param clazz the class being introspected
+     * @param methodName the method expected
+     * @param <T> the expected result. It is *not* checked
+     * @return a function that takes a {@link String} and produces a {@link T}
+     */
+    @SuppressWarnings("unchecked")
+    @Nullable
+    private static <T> Function<String, T> getPublicStaticMethod(final Class<T> clazz,
+                                                                 final String methodName) {
+        try {
+            final Method method = clazz.getMethod(methodName, String.class);
```

Comment:
- Could we skip the method that returns the type that's not compatible with `clazz`?
- How about using `MethodHandle`, which is faster than reflection? https://www.baeldung.com/java-method-handles

Reply:
uhm.. yes. 

Never heard of MethodHandle before. Looking into it

Reply:
Fixed